src/routes/Settings/SettingsIndexView.js

```diff
@@ -2,20 +2,147 @@
+  systemLanguageCode,
+  hasPinCode,
+  isFullMnemonic,
+  networkName,
+  walletName,
+  derivationPath,
+  passphrase,
+}): {...SettingsGridCardProps, searchTags: string}[] => [{
+  title: 'Local currency',
+  description: formatCurrency(localCurrencyCode),
+  path: 'settings/currency',
+  iconName: 'local-currency',
+  searchTags: '',
+}, {
+  title: 'Default GAS Price',
+  description: divideThousands(defaultGasPrice),
+  path: 'settings/gas-price',
+  iconName: 'time',
+  searchTags: '',
+}, {
+  title: 'System language',
+  description: formatLanguage(systemLanguageCode),
+  path: 'settings/language',
+  iconName: 'language',
+  searchTags: 'locale translation',
+}, {
+  title: 'PIN Code',
+  description: formatBoolean(hasPinCode),
+  path: 'settings/pin-code',
+  iconName: 'lock-pin',
+  searchTags: '',
+}, {
+  title: 'Security password',
+  description: 'Change',
+  path: 'settings/password',
+  iconName: 'lock-pin',
+  searchTags: '',
+}, {
+  title: 'Exchange service',
+  description: 'Try Jcash',
+  path: 'https://jcash.network/',
+  iconName: 'exchange-service',
+  searchTags: '',
+}, {
+  title: 'Support',
+  description: 'Send ticket to support',
+  path: 'settings/support',
+  iconName: 'message',
+  searchTags: '',
+}, {
+  title: 'Sing a message',
+  description: isFullMnemonic ? 'Enable' : ' ',
+  path: 'settings/sign',
+  iconName: 'message',
+  searchTags: '',
+}, {
+  title: 'Check a signature',
+  description: isFullMnemonic ? 'Enable' : ' ',
+  path: 'settings/check-signature',
+  iconName: 'protect',
+  iconColor: 'blue',
+  searchTags: '',
+}, {
+  title: 'Backup wallet',
+  description: 'Save your money!',
+  path: 'settings/backup',
+  iconName: 'backup-wallet',
+  searchTags: '',
+}, {
+  title: 'Network name',
+  description: networkName || ' ',
+  path: 'settings/network',
+  iconName: 'network',
+  searchTags: '',
+}, {
+  title: 'Rename wallet',
+  description: walletName,
+  path: 'settings/rename',
+  iconName: 'backup-wallet',
+  searchTags: '',
+}, {
+  title: 'Derivation path',
+  description: isFullMnemonic && derivationPath ? derivationPath : ' ',
+  path: 'settings/derivation',
+  iconName: 'setting',
+  searchTags: '',
+}, {
+  title: 'Passphrase',
+  description: isFullMnemonic && passphrase ? passphrase : ' ',
+  path: 'settings/passphrase',
+  iconName: 'setting',
+  searchTags: '',
+}, {
+  title: 'Delete wallet',
+  description: 'Badaaaah!',
+  path: 'settings/delete',
+  iconName: 'cross-circle',
+  iconColor: 'red',
+  searchTags: '',
+}]
+
+type State = {|
+  searchQuery: string
+|}
+
+class SettingsIndexView extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      searchQuery: '',
+    }
+  }
+
+  setSearchQuery = (query: string): void => {
+    this.setState({ searchQuery: query.toLowerCase() })
+  }
+
+  filterCardByQuery = (query: string): boolean => {
+    const queryPattern = new RegExp(`.*${this.state.searchQuery}.*`)
```

Comment:
This regexp is very inefficient. You don't need leading and trailing `.*`, but you need flags `g` and `i`.

Reply:
If you add `i`, you also don't need `toLowerCase` conversion.